machine_test.go

```diff
@@ -939,3 +944,50 @@ func TestPID(t *testing.T) {
 	}
 
 }
+
+func TestCaptureFifoToFileLeak(t *testing.T) {
+	m := &Machine{
+		exitCh: make(chan struct{}),
+	}
+
+	fifoPath := filepath.Join(testDataPath, "TestCaptureFifoToFileLeak.fifo")
+	err := syscall.Mkfifo(fifoPath, 0700)
+	require.NoError(t, err, "failed to make fifo")
+	defer os.Remove(fifoPath)
+
+	fd, err := syscall.Open(fifoPath, syscall.O_RDWR|syscall.O_NONBLOCK, 0600)
+	require.NoError(t, err, "failed to open fifo path")
+	f := os.NewFile(uintptr(fd), fifoPath)
+	assert.NotNil(t, f, "failed to create new  file")
+	go func() {
+		for {
+			select {
+			case <-m.exitCh:
+				break
+			default:
+				_, err := f.Write([]byte("A"))
+				assert.NoError(t, err, "failed to write bytes to fifo")
+			}
+		}
+	}()
+
+	buf := bytes.NewBuffer(nil)
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	loggerBuffer := bytes.NewBuffer(nil)
+	logger := fctesting.NewLogEntry(t)
+	logger.Logger.Level = logrus.WarnLevel
+	logger.Logger.Out = loggerBuffer
+	err = m.captureFifoToFile(ctx, logger, fifoPath, buf)
+	assert.NoError(t, err, "failed to capture fifo to file")
+	select {
+	case <-m.exitCh:
+	case <-ctx.Done():
+	}
```

Comment:
Doesn't it always go to ctx.Done() branch after 1 second? How about explicitly calling cancel() or put something on the channel instead of waiting?

Reply:
Yea, I can call cancel. Thanks!